daemon/image_delete.go

```diff
@@ -93,27 +96,38 @@ func (daemon *Daemon) imgDeleteHelper(name string, list *[]types.ImageDelete, fi
 		return nil
 	}
 
-	if len(repos) <= 1 || (len(repoAndTags) <= 1 && deleteByID) {
-		if err := daemon.canDeleteImage(img.ID, force); err != nil {
-			return err
+	// Untag the current image
+	untag := func(repoAndTags map[string][]string) error {
+		for repoName, tags := range repoAndTags {
+			for _, tag := range tags {
+				tagDeleted, err := daemon.Repositories().Delete(repoName, tag)
+				if err != nil {
+					return err
+				}
+				if tagDeleted {
```

Comment:
Here is an example of where better style can lead to clarity. The primary operation of this loop is to append a deletion to list. It should be in the main body:

```
if !tagDeleted {
    continue
}

*list = append(*list, types.ImageDelete{
    Untagged: utils.ImageReference(repoName, tag),
 })
daemon.EventsService.Log("untag", img.ID, "")
```